lib/responder.js

```diff
@@ -378,11 +379,25 @@ BodyDispatcher.prototype.promise = function dispatchBodyPromise(
 
   var isString = operation.copyResponseHeaders(response);
 
-  // concatStream accumulates response with callback
-  response.pipe(concatStream(
-      {encoding: (isString ? 'string' : 'buffer')},
-      collectObject
-      ));
+  if(response.headers['content-encoding']==='gzip') {
```

Comment:
I think a function is in order as this comparison appears 13 times now. `if (responseIsGzipped(response))` would read nicely and would eliminate the duplication. It could also easily check for "content-encoding" and "Content-Encoding" as well.